dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -57,6 +58,85 @@ static int iDriveTypeIds[7] = { IDS_DRIVE_FIXED,       /* DRIVE_UNKNOWN */
 *   IShellFolder implementation
 */
 
+static BOOL TryToLockDrive(HANDLE hDrive, DWORD dwRetryCount, DWORD dwSleep)
```

Comment:
You might want to make it a bit more generic to support both lock and unlock operations.
You may have a look at LockOrUnlockVolume() in fsutil tool: https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/cmdutils/fsutil/volume.c;hb=HEAD#l21